core/templates/pages/exploration-editor-page/improvements-tab/services/improvement-feedback-thread-modal.controller.spec.ts

```diff
@@ -0,0 +1,167 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for ImprovementFeedbackThreadModalController.
+ */
+
+describe('Improvement Feedback Thread Modal Controller', function() {
+  var $scope = null;
+  var $uibModalInstance = null;
+  var $httpBackend = null;
+  var $q;
+  var expId = 'exp1';
+  var AlertsService = null;
+  var ThreadDataService = null;
+  var CsrfService = null;
+  var thread = null;
+
+  beforeEach(angular.mock.module('oppia'));
+  beforeEach(angular.mock.module(function($provide) {
+    $provide.value('ExplorationDataService', {
+      explorationId: expId
+    });
+    $provide.value('UserService', {
+      getUserInfoAsync: function() {
+        return $q.resolve({
+          isLoggedIn: function() {
+            return true;
+          }
+        });
+      }
+    });
+  }));
+  beforeEach(angular.mock.inject(function($injector, $controller) {
+    AlertsService = $injector.get('AlertsService');
+    ThreadDataService = $injector.get('ThreadDataService');
+    CsrfService = $injector.get('CsrfTokenService');
+    $httpBackend = $injector.get('$httpBackend');
+    $q = $injector.get('$q');
+
+    spyOn(CsrfService, 'getTokenAsync').and.callFake(function() {
+      var deferred = $q.defer();
+      deferred.resolve('sample-csrf-token');
+      return deferred.promise;
+    });
+
+    var mockFeedbackThreads = [{
+      last_updated: 1000,
+      original_author_username: 'author',
+      status: 'accepted',
+      subject: 'sample subject',
+      summary: 'sample summary',
+      message_count: 10,
+      state_name: 'state 1',
+      thread_id: 'exp1.thread1'
+    }];
+
+    $httpBackend.whenGET('/threadlisthandler/' + expId)
+      .respond({
+        feedback_thread_dicts: mockFeedbackThreads,
+        suggestion_thread_dicts: []
+      });
+    $httpBackend.whenGET(
+      '/suggestionlisthandler?target_type=exploration&target_id=' + expId
+    ).respond({ suggestions: [] });
+    ThreadDataService.fetchThreads().then(function(threads) {
+      thread = threads.feedbackThreads[0];
+    });
+    $httpBackend.flush();
```

Comment:
add an after each block to make sure that there are no outstanding http requests.

Reply:
Done.